docs/data/material/migration/migrating-to-v6/migrating-to-v6.md

```diff
@@ -0,0 +1,263 @@
+# Migrating to v6
+
+<p class="description">This guide explains how and why to migrate from Material UI v5 to v6.</p>
+
+## Start using the alpha release
+
+In the `package.json` file, change the package version from `latest` to `next`.
+
+```diff title="package.json"
+-"@mui/material": "latest",
++"@mui/material": "next",
+```
+
+Optionally, if you are using one of these packages, be sure to change the version to `next` too:
+
+- `@mui/icons-material`
+- `@mui/system`
+- `@mui/lab`
+- `@mui/material-nextjs`
+- `@mui/styled-engine-sc`
+- `@mui/utils`
+
+Using `next` ensures your project always uses the latest v6 alpha release.
+Alternatively, you can also target and fix it to a specific version, for example, `6.0.0-alpha.0`.
+
+## Why you should migrate
+
+Material UI v6's biggest highlight is the inclusion of Pigment CSS, a next-gen zero-runtime CSS-in-JS library, as an opt-in styling engine.
+Using it will make your project compatible with React Server Components and reduce its bundle size due to the styles extraction, avoiding client-side recalculation.
+
+Aside from that, the CSS variables API for Material UI is now stable in v6.
+Relying on CSS variables allows for more intricate customization possibilities and better overall DX.
+
+A major feedback we've learned from the v5 cycle is to reduce the number of breaking changes to the smallest.
+Material UI v6 does exactly that: minimal breaking changes, including only browser support updates and a Node.js version bump.
+These updates can, for the most part, be migrated automatically via codemods.
+
+Last but not least, Material UI v6 also includes new features, such as container queries and a theme utility for styling different color schemes.
```

Comment:
```suggestion
Material UI v6's biggest highlight is the introduction of Pigment CSS, a next-gen zero-runtime CSS-in-JS library, as an opt-in styling engine.
Using it will make your project compatible with React Server Components, as well as help reduce its bundle size due to styles being extracted at build time, avoiding client-side recalculation.

As a lesson learned from v5, this major release introduces minimal breaking changes.
Namely, browser support updates, a Node.js version bump, and the removal of the UMD bundle.
These reduced the Material UI package size by 2.5MB, 25% of the total size.

These updates can be, for the most part, migrated automatically via codemods.

Aside from that, v6 also includes a few quality-of-life improvements regarding styling:
- The `CssVarsProvider` API is now stable. That enables you to rely on CSS variables, allowing for more intricate and performant customization possibilities, along with improved overall developer experience
- Support for container queries within the theme
- A new theme utility for adding styles to specific color modes
```

Refining this section a bit more—I felt like it could be polished, as well as mentioning the package size reduction. I'm looking forward to seeing what Sam has to say about this!